winsup/cygwin/include/cygwin/exit_process.h

```diff
@@ -116,33 +128,96 @@ kill_via_console_helper (HANDLE process, wchar_t *function_name, int exit_code,
 static int current_is_wow = -1;
 static int is_32_bit_os = -1;
 
+typedef BOOL (WINAPI * IsWow64Process2_t) (HANDLE, USHORT *, USHORT *);
+static bool wow64process2initialized = false;
+static IsWow64Process2_t pIsWow64Process2 /* = NULL */;
+
+typedef BOOL (WINAPI * GetProcessInformation_t) (HANDLE,
+                                                 PROCESS_INFORMATION_CLASS,
+                                                 LPVOID, DWORD);
+static bool getprocessinfoinitialized = false;
+static GetProcessInformation_t pGetProcessInformation /* = NULL */;
+
 static BOOL
-get_wow (HANDLE process, BOOL &is_wow, int &bitness)
+get_wow (HANDLE process, BOOL &is_wow, USHORT &process_arch)
 {
-  if (is_32_bit_os == -1)
+  USHORT native_arch = IMAGE_FILE_MACHINE_UNKNOWN;
+  if (!wow64process2initialized)
     {
-      SYSTEM_INFO info;
-
-      GetNativeSystemInfo (&info);
-      if (info.wProcessorArchitecture == 0)
-        is_32_bit_os = 1;
-      else if (info.wProcessorArchitecture == 9)
-        is_32_bit_os = 0;
-      else
-        is_32_bit_os = -2;
+      pIsWow64Process2 = (IsWow64Process2_t)GetProcAddress (
+                                              GetModuleHandle ("KERNEL32"),
+                                              "IsWow64Process2");
```

Comment:
Not sure about where to line up args in these cases...

Reply:
It looks like Cygwin wants to align function parameters to the column where the first parameter starts (using horizontal tabs and then spaces for the alignment), and not end in an open parenthesis. To avoid the latter, assignments seem to be wrapped after the `=` (and the next line gets an additional horizontal tab) or after a cast. In your case, I think it would look like this:

```c
	pIsWow64Process2 = (IsWow64Process2_t)
		GetProcAddress (GetModuleHandle ("KERNEL32"),
				"IsWow64Process2");
```

Reply:
Gotta make sure the important stuff is right :wink:

Reply:
Hah!